core/templates/pages/contributor-dashboard-page/modal-templates/translation-modal.component.ts

```diff
@@ -401,7 +401,41 @@ export class TranslationModalComponent {
     return this.translateTextService.getActiveIndex() > 0;
   }
 
-  skipActiveTranslation(): void {
+  getTrimmedTranslationContent(): string {
+    const content = Array.isArray(this.activeWrittenTranslation)
+      ? this.activeWrittenTranslation.join('')
+      : this.activeWrittenTranslation || '';
+    return content.trim();
+  }
+
+  confirmUnsavedChanges(isSaveDisabled: boolean): boolean {
+    if (isSaveDisabled) {
+      return true;
+    }
+    const translationContent = this.getTrimmedTranslationContent();
```

Comment:
This seems a bit excessive just for checking whether the content is empty or not. I think it can be done more directly.